### DISCUSSION TITLE
data propagation (layout->pages)

### DISCUSSION THREAD

**Discussion by vorcigernix** [sveltejs/kit#7292] [Q&A] [answered]
Hi, 
I must have missed something from the basics of data fetching logic. The way I understand the documentation, I can work with the data in the following way:

- load everything in +layout.js
- filter in +page.js
- display in +page.svelte

Now, it seems to be working to the +page.js where I can read data through `await parent()` - but the data that are displayed in +page.svelte are still the data from layout. What could be wrong? Is it supposed to work this way? 

Code is here: https://github.com/ethbrno/eb2-website/tree/dev/src/routes/contributors/%5Btype%5D

**Comment by dummdidumm** [comment, marked as answer]
You are returning `{ data: contributors }`. SvelteKit merges the data from the above layouts into the final output, which is then `{ data: .., contributors: .., .. }`. I think you want to return `{ contributors: contributors }` (or `{ contributors }`) instead, then the data from the layout is properly overridden.

**Reply by vorcigernix** [reply]
Thanks so much. Is there a way to throw away the original data object completely? I mean, I need it only as a base for the data I am filtering for the page, so they don't need to be passed. It is a theoretical question, it does not bother me considering a size of the data.

**Reply by dummdidumm** [reply]
You could do `{ contributors: []` } to override it in the page - but since it's used in the layout anyway, it doesn't make any difference with regards to the size of the response.

**Comment by vorcigernix** [comment]
Haven't realized that, thanks again :-)
